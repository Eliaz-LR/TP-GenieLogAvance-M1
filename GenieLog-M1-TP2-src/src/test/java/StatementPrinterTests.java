import static org.approvaltests.Approvals.verify;

import java.util.List;
import org.junit.jupiter.api.Test;

public class StatementPrinterTests {

  @Test
  void exampleStatement() {
    Invoice invoice = new Invoice(
      "BigCo",
      List.of(
        new Performance(new Play("Hamlet", Play.PlayType.TRAGEDY), 55),
        new Performance(new Play("As You Like It", Play.PlayType.COMEDY), 35),
        new Performance(new Play("Othello", Play.PlayType.TRAGEDY), 40)
      )
    );

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.toText(invoice);

    verify(result);
  }

  @Test
  void htmlStatement() {
    Invoice invoice = new Invoice(
      "BigCo",
      List.of(
        new Performance(new Play("Hamlet", Play.PlayType.TRAGEDY), 55),
        new Performance(new Play("As You Like It", Play.PlayType.COMEDY), 35),
        new Performance(new Play("Othello", Play.PlayType.TRAGEDY), 40)
      )
    );

    StatementPrinter statementPrinter = new StatementPrinter();
    System.out.println(statementPrinter.toHtml(invoice));
  }
}
