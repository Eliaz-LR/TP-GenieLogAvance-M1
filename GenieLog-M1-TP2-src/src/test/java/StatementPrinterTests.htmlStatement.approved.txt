<html>
<head>
  <title>Invoice</title>
</head>
<body>
  <h1>Invoice</h1>
  <ul>
    <li><b>Client :</b> BigCo</li>
  </ul>
  <table>
    <thead>
        <tr>
            <th colspan="4"> Invoice Table </th>
        </tr>
        <tr>
            <td class="categorie">Piece</td>
            <td class="categorie">Seats sold</td>
            <td class="categorie">Price</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Hamlet</td>
            <td>55</td>
            <td>650,00 €</td>
        </tr>
        <tr>
            <td>As You Like It</td>
            <td>35</td>
            <td>580,00 €</td>
        </tr>
        <tr>
            <td>Othello</td>
            <td>40</td>
            <td>500,00 €</td>
        </tr>
        <tr>
            <td class="endTab" colspan="2">Total owed:</td>
            <td >1 730,00 €</td>
        </tr>
        <tr>
            <td class="endTab" colspan="2">Fidelity points earned: </td>
            <td >47</td>
        </tr>
    </tbody>
  </table>
  <p id="endMsg">Payment is required under 30 days.</p>
</body>
</html>
